### PR TITLE
User Story 5: Delete Button

### PR DESCRIPTION
**Pull request by JoergWolff**

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -45,6 +45,20 @@ export default function App({ Component, pageProps }) {
...
+    router.push("/");
+  }
+
+  function handleDeleteClick(isClicked, currentBook) {
+    if (isClicked) {
+      handleDeleteBook(currentBook);
+    } else {
+      router.push("/");
+    }
+  }
```
Comment (F-Kirchhoff) [inline pages/_app.js:60]: here you can simply refactor the router.push into the handleCancel function. Then hand over the handleDeleteBook to the onDelete prop of the modal and the handleCancel to the onCancel prop

File: pages/detail/book/[id]/detail.js [hunk 2]
```diff
@@ -13,9 +22,25 @@ export default function DetailBookPage({ data }) {
...
+
   return (
     <StyledDetail>
-      <Card book={currentBook} />
+      <ModalDelete
+        book={currentBook}
+        onDeleteClick={onDeleteClick}
+        isVisible={isVisible}
+        onToggleVisible={toggleVisible}
+      />
```
Comment (F-Kirchhoff) [inline pages/detail/book/[id]/detail.js:33]: If you encounter problems with the layering of the modal above all other content, I suggest you to use React.createPortal. This is a method to hook parts of your app not into the #root element on your index.html but into a separate container, for example a #modalWrapper element. By that, you will never have problems displaying the modal above other content. But if it works for now, you can leave it as it is.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Jörg, your code looks good, there is only one concept that I highlighted regarding the different actions of your modal. I think it will simplify your modal. Have a look!

File: components/ModalDelete/ModalDelete.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  if (isVisible) {
+    return (
+      <StyledModal>
+        <StyledModalContent>
+          <h2>Are you sure to delete:</h2>
+          <h3>{book?.author}</h3>
+          <h4>{book?.title}</h4>
+          <StyledModalButton onClick={() => onDeleteClick(false, book)}>
+            Cancel
+          </StyledModalButton>
```
[the remark this answers is not in the document]
Comment (JoergWolff) [reply components/ModalDelete/ModalDelete.js]: I passed the props onDelete and onCancel to the Delete.js and passed them to the respective buttons.

File: pages/_app.js (hunk 1 above)
Comment (JoergWolff) [reply pages/_app.js]: Changes:
```  function handleDeleteBook(currentBook) {
    setBooks(books.filter((book) => book.id !== currentBook.id));

    router.push("/");
  }
```
and
```
  function handleCancel() {
    router.push("/");
  }
´´´
to pass them to props.

File: pages/detail/book/[id]/detail.js (hunk 2 above)
Comment (JoergWolff) [reply pages/detail/book/[id]/detail.js]: Maybe I'll change it again later.

**Review by marcterre — APPROVED** [review]
Hey Jörg, nice small pr. I like how structured and well readable your code is. For me it looks good. 🙂

**Pull request merged**